src/js/component.js

```diff
@@ -17,31 +17,50 @@ import mergeOptions from './utils/merge-options.js';
  * Components are embeddable UI objects that are represented by both a
  * javascript object and an element in the DOM. They can be children of other
  * components, and can have many children themselves.
- * ```js
- *     // adding a button to the player
- *     var button = player.addChild('button');
- *     button.el(); // -> button element
+ *
+ * Creating a button component
+ * ``` js
+ *   // adding a button to the player
+ *   var player = videojs('some-video-id');
+ *   var Component = videojs.getComponent('Component');
+ *   var button = new Component(player);
+ *
+ *   console.log(button.el());
  * ```
+ *
+ * Above code will log this html
  * ```html
- *     <div class="video-js">
- *       <div class="vjs-button">Button</div>
- *     </div>
+ *   <div class="video-js">
+ *     <div class="vjs-button">Button</div>
+ *   </div>
  * ```
- * Components are also event targets.
- * ```js
- *     button.on('click', function() {
- *       console.log('Button Clicked!');
- *     });
- *     button.trigger('customevent');
+ *
+ * Adding a button to the player
+ * ``` js
+ *   // adding a button to the player
+ *   var player = videojs('some-video-id');
+ *   var button = player.addChild('button');
+ *
+ *   console.log(button.el());
+ *   // will have the same html result as the previous example
```

Comment:
A part of me feels like we need to have doc comments that link to guides, and the guides should be the ones with the examples. Maybe another PR to rip examples out into manual documentation at a later date?